tests/acceptance/05_processes/process_stop.cf

```diff
@@ -0,0 +1,148 @@
+body common control
+{
+    inputs => { "../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+}
+
+bundle agent test
+{
+  meta:
+      "description" -> { "ENT-4988" }
+        string => "Test some basic expectations when using process_stop in processes type promises";
+
+  processes:
+
+      # The policy file itself is not expected to be executable, this promise is
+      # expected to fail and be notkept.
+
+      "."
+        process_stop => "$(this.promise_filename)",
+        handle => "process_stop_not_executable_expect_failed",
+        classes => explicit_results( "namespace", "$(this.handle)_is" );
+
+      # G.true is expected to return true, and the promise is expected to be
+      # repaired. Note: At the time of authorship there is no validation that
+      # the selected pids were killed by process_stop. We are only using the
+      # return code.
+
+      "."
+        process_stop => "$(G.true)",
+        handle => "process_stop_return_zero_expect_repaired",
+        classes => explicit_results( "namespace", "$(this.handle)_is" );
+
+      # G.false is expected to return true, and the promise is expected to be
+      # repaired. Note: At the time of authorship there is no validation that
+      # the selected pids were killed by process_stop. We are only using the
+      # return code.
+
+      "."
+        process_stop => "$(G.false)",
+        handle => "process_stop_return_nonzero_expect_failed",
+        classes => explicit_results( "namespace", "$(this.handle)_is" );
+
+      # G.echo is expected to return true, and the promise is expected to be
+      # repaired. Note: At the time of authorship there is no validation that
+      # the selected pids were killed by process_stop. We are only using the
+      # return code.
+
+      "."
+        process_stop => "$(G.echo) pretend stop servicename",
+        handle => "process_stop_with_args_return_nonzero_expect_repaired",
+        classes => explicit_results( "namespace", "$(this.handle)_is" );
+}
+
+bundle agent check
+{
+  vars:
+
+      "expected_classes" slist => {
+                    "process_stop_with_args_return_nonzero_expect_repaired_is_repaired",
+                    "process_stop_return_nonzero_expect_failed_is_failed",
+                    "process_stop_return_zero_expect_repaired_is_repaired",
+                    "process_stop_not_executable_expect_failed_is_failed",
+      };
+
+
+      DEBUG::
+      "found_classes" slist => classesmatching( "process_stop_.*");
+      "difference" slist => difference( found_classes, expected_classes );
+
+  classes:
+      "ok" and => { @(expected_classes) };
+
+  reports:
+    DEBUG::
+      "Found unexpected class: $(difference)";
+
+    ok::
+      "$(this.promise_filename) Pass";
+    !ok::
+      "$(this.promise_filename) FAIL";
+}
+
+body classes explicit_results(scope, class_prefix)
```

Comment:
Is it possible to share this between tests?

Reply:
Not from where it currently is. Usually, I use results, which is in the stdlib, but it defined more classes than I wanted in this case, so I made this custom body.

If you want this in dcs lib, then i can do that in a separate pr.